src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountExtractPdpEntryServiceImpl.java

```diff
@@ -0,0 +1,167 @@
+
+    @Override
+    public PdpFeedHeaderEntry buildPdpFeedHeaderEntry(Date batchDate) {
+        PdpFeedHeaderEntry headerEntry = new PdpFeedHeaderEntry();
+        headerEntry.setChart(getConcurParamterValue(ConcurConstants.CUSTOMER_PROFILE_LOCATION_PARAMETER_NAME));
+        headerEntry.setCreationDate(formatDate(batchDate));
+        headerEntry.setSubUnit(getConcurParamterValue(ConcurConstants.CUSTOMER_PROFILE_SUB_UNIT_PARAMETER_NAME));
+        headerEntry.setUnit(getConcurParamterValue(ConcurConstants.CUSTOMER_PROFILE_UNIT_PARAMETER_NAME));
+        return headerEntry;
+    }
+
+    @Override
+    public PdpFeedGroupEntry buildPdpFeedGroupEntry(ConcurStandardAccountingExtractDetailLine line) {
+        PdpFeedGroupEntry currentGroup = new PdpFeedGroupEntry();
+        currentGroup.setPayeeName(buildPayeeName(line.getEmployeeLastName(), line.getEmployeeFirstName(), 
+                line.getEmployeeMiddleInitital()));
+        currentGroup.setPayeeId(buildPayeeIdEntry(line));
+        currentGroup.setPaymentDate(formatDate(line.getBatchDate()));
+        currentGroup.setCombineGroupInd(ConcurConstants.StandardAccountingExtractPdpConstants.COMBINED_GROUP_INDICATOR);
+        currentGroup.setBankCode(ConcurConstants.StandardAccountingExtractPdpConstants.BANK_CODE);
+        currentGroup.setCustomerInstitutionIdentifier(StringUtils.EMPTY);
+        return currentGroup;
+    }
+    
+    protected String buildPayeeName(String lastName, String firstName, String middleInitial) {
+        String separator = KFSConstants.COMMA + KFSConstants.BLANK_SPACE;
+        String fullName = lastName + separator + firstName;
+        if (StringUtils.isNotBlank(middleInitial)) {
+            fullName = fullName + separator + middleInitial + KFSConstants.DELIMITER;
+        }
+        if(fullName.length() > getPayeeNameFieldSize()) {
+            fullName = StringUtils.substring(fullName, 0, getPayeeNameFieldSize());
+            fullName = removeLastCharacterWhenComma(fullName);
+        }
+        return fullName;
+    }
+    
+    private String removeLastCharacterWhenComma(String fullName) {
+        if (fullName.lastIndexOf(KFSConstants.COMMA) >= getPayeeNameFieldSize()-2) {
+            fullName = fullName.substring(0, fullName.lastIndexOf(KFSConstants.COMMA));
+        }
+        return fullName;
+    }
+
+    @Override
+    public PdpFeedPayeeIdEntry buildPayeeIdEntry(ConcurStandardAccountingExtractDetailLine line) {
+        PdpFeedPayeeIdEntry payeeIdEntry = new PdpFeedPayeeIdEntry();
+        payeeIdEntry.setContent(line.getEmployeeId());
+        if (StringUtils.equalsIgnoreCase(line.getEmployeeStatus(), ConcurConstants.StandardAccountingExtractPdpConstants.EMPLOYEE_STATUS_CODE)) {
+            payeeIdEntry.setIdType(ConcurConstants.StandardAccountingExtractPdpConstants.EMPLOYEE_PAYEE_STATUS_TYPE_CODE);
+        } else {
+            payeeIdEntry.setIdType(ConcurConstants.StandardAccountingExtractPdpConstants.NON_EMPLOYEE_PAYEE_STATUS_TYPE_CODE);
```

Comment:
You are assuming that all other values not equal to EMPLOYEE are to be set to Y. The spec for both SAE and request extract indicates that the data value of "NON-EMPLOYEE" should be set to Y. To me this implies there could be other values sent from Concur in that column besides EMPLOYEE and NON-EMPLOYEE. Thoughts?